File: hw/quina/u3/hw29singleton/AlertSystem/AlarmService/src/main/java/ec/edu/espe/alarmservice/utils/MongoConnection.java
package ec.edu.espe.alarmservice.utils;

import com.mongodb.client.MongoClient;
import com.mongodb.client.MongoClients;
import com.mongodb.client.MongoDatabase;
/**
 *
 * @author Maryuri Quiña, @ESPE
 */
public class MongoConnection {

    private static final String URI = "mongodb+srv://maryuri:maryuri2007@cluster0.iektq66.mongodb.net/";
    private static MongoClient client;

    public static MongoDatabase getDatabase() {
        if (client == null) {
            client = MongoClients.create(URI);
        }
        return client.getDatabase("InventoryDB");
    }
}

File: hw/quina/u3/hw29singleton/AlertSystemPy/utils/mongo_connection.py
from pymongo import MongoClient

class MongoConnection:
    _client = None
    _URI = "mongodb+srv://maryuri:maryuri2007@cluster0.iektq66.mongodb.net/"

    @staticmethod
    def get_database():
        if MongoConnection._client is None:
            MongoConnection._client = MongoClient(MongoConnection._URI)
        return MongoConnection._client["InventoryDB"]